bundles/org.openhab.binding.bmwconnecteddrive/src/main/java/org/openhab/binding/bmwconnecteddrive/internal/handler/VehicleChannelHandler.java

```diff
@@ -0,0 +1,646 @@
+import org.openhab.binding.bmwconnecteddrive.internal.dto.charge.WeeklyPlanner;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.statistics.AllTrips;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.statistics.LastTrip;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.status.CBSMessage;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.status.CCMMessage;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.status.Doors;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.status.Position;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.status.VehicleStatus;
+import org.openhab.binding.bmwconnecteddrive.internal.dto.status.Windows;
+import org.openhab.binding.bmwconnecteddrive.internal.utils.Constants;
+import org.openhab.binding.bmwconnecteddrive.internal.utils.Converter;
+import org.openhab.binding.bmwconnecteddrive.internal.utils.VehicleStatusUtils;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VehicleChannelHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class VehicleChannelHandler extends BaseThingHandler {
+    protected final Logger logger = LoggerFactory.getLogger(VehicleChannelHandler.class);
+    protected boolean imperial = false;
+    protected boolean hasFuel = false;
+    protected boolean isElectric = false;
+    protected boolean isHybrid = false;
+
+    // List Interfaces
+    protected List<CBSMessage> serviceList = new ArrayList<CBSMessage>();
+    protected String selectedService = Constants.UNDEF;
+    protected List<CCMMessage> checkControlList = new ArrayList<CCMMessage>();
+    protected String selectedCC = Constants.UNDEF;
+    protected List<Destination> destinationList = new ArrayList<Destination>();
+    protected String selectedDestination = Constants.UNDEF;
+
+    protected BMWConnectedDriveOptionProvider optionProvider;
+
+    // Vehicle Status Channels
+    protected ChannelUID doors;
+    protected ChannelUID windows;
+    protected ChannelUID lock;
+    protected ChannelUID serviceNextDate;
+    protected ChannelUID serviceNextMileage;
+    protected ChannelUID checkControl;
```

Comment:
You can easily create the ChannelUIDs on demand. So it seems pretty wasteful to make fields for each of them. Please remove these fields.

Reply:
fixed

Reply:
Channel UID creation is taking for each update several String concatenations. So I found it pretty watseful to cerate these on demand. But fine for me.